### ISSUE TITLE
OptionBool::ok_or_else doctest fails on beta (1.20)

### ISSUE THREAD

**Issue by CAD97** [llogiq/optional#20]
```
---- src/lib.rs - OptionBool::ok_or_else (line 422) stdout ----
	error: code relies on type inference rules which are likely to change
 --> <anon>:4:2
  |
4 |  assert_eq!(OptionBool::SomeTrue.ok_or_else(|| panic!()), Ok(true));
  |  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: #[deny(resolve_trait_on_defaulted_unit)] on by default
  = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
  = note: for more information, see issue #39216 <https://github.com/rust-lang/rust/issues/39216>
```

**Comment by CAD97** [comment]
See also https://github.com/rust-lang/rust/issues/39216#issuecomment-284482357

**Comment by CAD97** [comment]
Fixed by #21

**Comment by llogiq** [comment]
Ah, thank you, I forgot to close this. :+1:
